test/micro/org/openjdk/bench/java/lang/foreign/AllocTest.java

```diff
@@ -66,45 +66,27 @@ private static class CallocArena implements Arena {
 
         static final MethodHandle CALLOC;
         static final MethodHandle FREE;
-        static final long ADDRESS_SIZE;
         static final Consumer<MemorySegment> CLEANUP;
 
         static {
-            ADDRESS_SIZE = AddressLayout.ADDRESS.byteSize();
             var linker = Linker.nativeLinker();
             var lookup = linker.defaultLookup();
             var callocAddr = lookup.findOrThrow("calloc");
             var freeAddr = lookup.findOrThrow("free");
-            if (ADDRESS_SIZE == Long.BYTES) {
-                CALLOC = linker.downcallHandle(callocAddr, FunctionDescriptor.of(ValueLayout.JAVA_LONG, ValueLayout.JAVA_LONG, ValueLayout.JAVA_LONG));
-                FREE = linker.downcallHandle(freeAddr, FunctionDescriptor.ofVoid(ValueLayout.JAVA_LONG));
-                CLEANUP = ms -> {
-                    try {
-                        FREE.invokeExact(ms.address());
-                    } catch (Throwable e) {
-                        throw new AssertionError(e);
-                    }
-                };
-            } else {
-                CALLOC = linker.downcallHandle(callocAddr, FunctionDescriptor.of(ValueLayout.JAVA_INT, ValueLayout.JAVA_INT, ValueLayout.JAVA_INT));
-                FREE = linker.downcallHandle(freeAddr, FunctionDescriptor.ofVoid(ValueLayout.JAVA_INT));
-                CLEANUP = ms -> {
-                    try {
-                        FREE.invokeExact((int)ms.address());
-                    } catch (Throwable e) {
-                        throw new AssertionError(e);
-                    }
-                };
-            }
+            CALLOC = linker.downcallHandle(callocAddr, FunctionDescriptor.of(ValueLayout.JAVA_LONG, ValueLayout.JAVA_LONG, ValueLayout.JAVA_LONG));
+            FREE = linker.downcallHandle(freeAddr, FunctionDescriptor.ofVoid(ValueLayout.JAVA_LONG));
```

Comment:
I'm not sure these changes are needed. It seems the main goal here is to avoid the cost of the capture associated with `CLayouts::freeMemory` ? If so, can't we just store that consumer into a static final and call it a day? I think then we could avoid the `static` init, and leave most of the code unchanged, except for the additional static field?